dll/win32/syssetup/wizard.c

```diff
@@ -411,6 +414,123 @@ static const PRODUCT_OPTION_DATA s_ProductOptionData[] =
     { L"\0", L"WinNT", 1, 0x300, 1 }
 };
 
+static const WCHAR* s_DefaultSoundEvents[][2] = 
+{
+    { L".Default", L"%SystemRoot%\\Media\\ReactOS_Default.wav" },
+    { L"AppGPFault", L"" },
+    { L"Close", L"" },
+    { L"CriticalBatteryAlarm", L"%SystemRoot%\\Media\\ReactOS_Battery_Critical.wav" },
+    { L"DeviceConnect",  L"%SystemRoot%\\Media\\ReactOS_Hardware_Insert.wav" },
+    { L"DeviceDisconnect", L"%SystemRoot%\\Media\\ReactOS_Hardware_Remove.wav" },
+    { L"DeviceFail", L"%SystemRoot%\\Media\\ReactOS_Hardware_Fail.wav" },
+    { L"LowBatteryAlarm", L"%SystemRoot%\\Media\\ReactOS_Battery_Low.wav" },
+    { L"MailBeep", L"%SystemRoot%\\Media\\ReactOS_Notify.wav" },
+    { L"Maximize", L"%SystemRoot%\\Media\\ReactOS_Restore.wav" },
+    { L"MenuCommand", L"%SystemRoot%\\Media\\ReactOS_Menu_Command.wav" },
+    { L"MenuPopup", L"" },
+    { L"Minimize", L"%SystemRoot%\\Media\\ReactOS_Minimize.wav" },
+    { L"Open", L"" },
+    { L"PrintComplete", L"%SystemRoot%\\Media\\ReactOS_Print_Complete.wav" },
+    { L"RestoreDown", L"" },
+    { L"RestoreUp", L"" },
+    { L"SystemAsterisk", L"%SystemRoot%\\Media\\ReactOS_Ding.wav" },
+    { L"SystemExclamation", L"%SystemRoot%\\Media\\ReactOS_Exclamation.wav" },
+    { L"SystemExit", L"%SystemRoot%\\Media\\ReactOS_Shutdown.wav" },
+    { L"SystemHand", L"%SystemRoot%\\Media\\ReactOS_Critical_Stop.wav" },
+    { L"SystemNotification", L"%SystemRoot%\\Media\\ReactOS_Balloon.wav" },
+    { L"SystemQuestion", L"%SystemRoot%\\Media\\ReactOS_Ding.wav" },
+    { L"SystemStart", L"%SystemRoot%\\Media\\ReactOS_Startup.wav" },
+    { L"WindowsLogoff", L"%SystemRoot%\\Media\\ReactOS_LogOff.wav" }
+/* Logon sound is already set by default for both Server and Workstation */
+};
+
+static const WCHAR* s_ExplorerSoundEvents[][2] = 
+{
+    { L"EmptyRecycleBin", L"%SystemRoot%\\Media\\ReactOS_Recycle.wav" },
+    { L"Navigating", L"%SystemRoot%\\Media\\ReactOS_Start.wav" }
+};
+
+static BOOL
+DoWriteSoundEvents(HKEY hKey,
+                   LPCWSTR lpSubkey,
+                   LPCWSTR lpValue[],
+                   DWORD dwSize)
+{
+    HKEY hRootKey, hEventKey, hDefaultKey;
+    LONG error;
+    ULONG i;
+    LPCWSTR pszData, pszDest[MAX_PATH][2];
+    DWORD cbData;
+
+    for (i = 0; i < dwSize; i++)
+    {
+        /* Expand environment variable from string */
+        if (!ExpandEnvironmentStringsW(&lpValue[i][1], (LPWSTR)pszDest, _countof(pszDest)))
+        {
+            /* Failed to expand, continue with the other files */
+            continue;
+        }
+
+        /* Check if the sound file exists and isn't a directory */
+        if (GetFileAttributesW(pszDest[i][1]) == INVALID_FILE_ATTRIBUTES ||
+            GetFileAttributesW(pszDest[i][1]) == FILE_ATTRIBUTE_DIRECTORY)
+        {
+            /* It does not, just continue with the other files */
+            continue;
+        }
+    }
```

Comment:
I suggested to move it into the other loop.... You have the final choice (but in any case, please fix that `ExpandEnvironmentStringsW` call and the `GetFileAttributesW` as I did).
If you want to keep it separate as it originally is, I would strongly suggest you add a comment before this for-loop saying something along the lines of: `Verify that the sound files exist and are actual files`.
(Because we don't actually care about the file expansion for what you're going to write in the registry below...)

Reply:
OK, I'm not against of your suggestion. On the contrary, it will look more compactly (if it will take less lines of the code). :slightly_smiling_face: